GPT2/data.py

```diff
@@ -0,0 +1,99 @@
+"""
+Module to deal with loading text data and sampling from it.
+"""
+import glob
+import numpy as np
+import os
+import tqdm
+
+
+def load_dataset(enc, path, combine):
+    paths = []
+
+    # Simple file
+    if os.path.isfile(path):
+        paths.append(path)
+
+    # Directory
+    elif os.path.isdir(path):
+        for (dirpath, _, fnames) in os.walk(path):
+            for fname in fnames:
+                paths.append(os.path.join(dirpath, fname))
+
+    # Assume glob
+    else:
+        paths = glob.glob(path)
+
+    # filter paths
+    paths = [p for p in paths if '.DS_Store' not in p]
+
+    token_chunks = []
+    raw_text = ''
+    for path in tqdm.tqdm(paths):
+
+        if path.endswith('.npz'):
+
+            # Pre-encoded
+            with np.load(path) as npz:
+                for item in npz.files:
+                    token_chunks.append(npz[item])
+        else:
+
+            # Plain text
+            with open(path, 'r') as fp:
```

Comment:
```python
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 43: ordinal not in range(128)
```
UnicodeDecodeError is occured in here. I think open with `utf-8` is right.

Reply:
Ok I think that would be better, thanks.